lib/client.js

```diff
@@ -553,6 +563,29 @@ Client.prototype.waitForSchemaAgreement = function (connection, callback) {
 };
 
 /**
+ * Waits for schema agreements and schedules schema metadata refresh.
+ * @param {Connection} connection
+ * @param event
+ * @param {Function} callback
+ * @ignore
+ * @internal
+ */
+Client.prototype.handleSchemaChange = function (connection, event, callback) {
+  var self = this;
+  this._waitForSchemaAgreement(connection, function (err) {
+    if (err) {
+      //we issue a warning but we continue with the normal flow
+      self.log('warning', 'There was an error while waiting for the schema agreement between nodes', err);
+    }
+    if (self.options.isMetadataEnabled) {
+      // schedule metadata refresh but do not wait for it to finish (no callback)
+      self.controlConnection.handleSchemaChange(event);
```

Comment:
Two possible issues here:

1. if the schema does not agree, it will still call `handleSchemaChange` which will be invoked on the control connection which will debounce and fetch the schema on the control connection.  Its possible that the control connection has a stale view of the data as the schema change wasn't fully propagated to it yet.
2. The callback is invoked because the metadata is updated via handleSchemaChange, so if someone goes to grab the metadata immediately after their request completes they may not get the most recent data yet, as it hasn't been fetched yet (still in the debouncer).